src/zeit/cms/relation/corehandlers.py

```diff
@@ -29,12 +30,30 @@ def update_referencing_objects_handler(context, event):
     if event.publishing:
         return
     # prevent recursion
-    if not gocept.async.is_async():
-        update_referencing_objects(context)
+    for entry in inspect.stack():
+        if entry[3] == 'update_referencing_objects':
+            return
```

Comment:
Wir haben uns für `inspect` entschieden, da es sowohl synchron als auch asynchron eine Rekursion vermeidet. Dies half auch beim testen, da Celery in der Testumgebung immer synchron ausgeführt wird.

Wenn `inspect` inakzeptable ist, wäre die richtige Lösung `update_referencing_objects.reques.id is None` zu prüfen. Der Task stellt nämlich über das `request` Objekt bei asynchroner Ausführung seine Job ID (und andere Metadaten) bereit. Leider nur bei asynchroner Ausführung, in der Testumgebung bleibt das `request` leer.